lib/model/model.go

```diff
@@ -566,32 +573,32 @@ func (m *Model) folderSharedWith(folder string, deviceID protocol.DeviceID) bool
 
 func (m *Model) ClusterConfig(deviceID protocol.DeviceID, cm protocol.ClusterConfigMessage) {
 	m.pmut.Lock()
-	if cm.ClientName == "syncthing" {
-		m.deviceVer[deviceID] = cm.ClientVersion
-	} else {
-		m.deviceVer[deviceID] = cm.ClientName + " " + cm.ClientVersion
-	}
-
-	event := map[string]string{
-		"id":            deviceID.String(),
-		"deviceName":    cm.DeviceName,
-		"clientName":    cm.ClientName,
-		"clientVersion": cm.ClientVersion,
-	}
+	_, exists := m.config[deviceID]
+	m.config[deviceID] = cm
+	m.pmut.Unlock()
 
-	if conn, ok := m.conn[deviceID]; ok {
-		event["type"] = conn.Type.String()
-		addr := conn.RemoteAddr()
-		if addr != nil {
-			event["addr"] = addr.String()
+	// Only fire off event and print the line if it's the first ClusterConfig
+	// we are seeing.
```

Comment:
General question; do we now send more than one clusterconfig per connection, or when would this happen?